extensions/cloudcredentials/vsphere/create.go

```diff
@@ -27,3 +27,11 @@ func CreateVsphereCloudCredentials(rancherClient *rancher.Client) (*cloudcredent
 	}
 	return resp, nil
 }
+
+func GetVspherePassword() string {
```

Comment:
We need a go docs comment here

Reply:
the password is not actually used in the config, rather it is passed to a secret and the secret is referenced on the object. So when you 'create' the cloud credential, the password isn't viewable or retrievable directly from the object from rancher. With this new function, I can retrieve the password from the Go object instead.

Reply:
So then why is it reading from the config. that bit is confusing

Reply:
updated the comment, yes this is reading from the config not from a go object. Because we don't return the go object in the CreateVsphereCloudCredentials function, there's no way to get it from there. So I have to read from the config again.

Reply:
Awwww! I get it sorry, this is the cloud cred not the node template proceed lol